streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableRepartitionMap.java

```diff
@@ -91,37 +94,37 @@ public void process(final K key, final Change<V> change) {
             // if the selected repartition key or value is null, skip
             // forward oldPair first, to be consistent with reduce and aggregate
             if (oldPair != null && oldPair.key != null && oldPair.value != null) {
-                context().forward(oldPair.key, new Change<>(null, oldPair.value));
+                context.forward(oldPair.key, new Change<>(null, oldPair.value));
             }
 
             if (newPair != null && newPair.key != null && newPair.value != null) {
-                context().forward(newPair.key, new Change<>(newPair.value, null));
+                context.forward(newPair.key, new Change<>(newPair.value, null));
             }
 
         }
     }
 
-    private class KTableMapValueGetter implements KTableValueGetter<K, KeyValue<K1, V1>> {
+    private class KTableMapValueGetter implements KTableValueGetter<K1, V1> {
 
         private final KTableValueGetter<K, V> parentGetter;
-        private ProcessorContext context;
 
         KTableMapValueGetter(final KTableValueGetter<K, V> parentGetter) {
             this.parentGetter = parentGetter;
         }
 
         @Override
-        public void init(final ProcessorContext context) {
-            this.context = context;
+        public void init(final ProcessorContext<Void, Void> context) {
             parentGetter.init(context);
         }
 
         @Override
-        public ValueAndTimestamp<KeyValue<K1, V1>> get(final K key) {
-            final ValueAndTimestamp<V> valueAndTimestamp = parentGetter.get(key);
-            return ValueAndTimestamp.make(
-                mapper.apply(key, getValueOrNull(valueAndTimestamp)),
-                valueAndTimestamp == null ? context.timestamp() : valueAndTimestamp.timestamp());
+        public ValueAndTimestamp<V1> get(final K1 key) {
+            // FIXME: we've been querying the wrong key this whole time.
+//            final ValueAndTimestamp<V> valueAndTimestamp = parentGetter.get(key);
+//            return ValueAndTimestamp.make(
+//                mapper.apply(key, getValueOrNull(valueAndTimestamp)),
+//                valueAndTimestamp == null ? context.timestamp() : valueAndTimestamp.timestamp());
+            throw new UnsupportedOperationException("FIXME");
```

Comment:
This is pretty significant... We probably need to just materialize the result of the Map when we need the valuegetter, just like we do with KTableSources.